app/controllers/nonresident/CalculatorController.scala

```diff
@@ -162,7 +162,88 @@ trait CalculatorController extends BaseController {
+    TotalGainModel(flatGain, rebasedGain, timeApportionedGain)
+  }
+
+  def calculateTotalGain(disposalValue: Double,
+                         disposalCosts: Double,
+                         acquisitionValue: Double,
+                         acquisitionCosts: Double,
+                         improvements: Double,
+                         rebasedValue: Option[Double],
+                         rebasedCosts: Double,
+                         disposalDate: Option[DateTime],
+                         acquisitionDate: Option[DateTime],
+                         improvementsAfterTaxStarted: Double): Action[AnyContent] = Action.async { implicit request =>
+
+    val result = calculateTotalGainModel(disposalValue,
+                                         disposalCosts,
+                                         acquisitionValue,
+                                         acquisitionCosts,
+                                         improvements,
+                                         rebasedValue,
+                                         rebasedCosts,
+                                         disposalDate,
+                                         acquisitionDate,
+                                         improvementsAfterTaxStarted)
+
+    Future.successful(Ok(Json.toJson(result)))
+  }
+
+  def calculateTaxableGainAfterPRR(disposalValue: Double,
+                                   disposalCosts: Double,
+                                   acquisitionValue: Double,
+                                   acquisitionCosts: Double,
+                                   improvements: Double,
+                                   rebasedValue: Option[Double],
+                                   rebasedCosts: Double,
+                                   disposalDate: Option[DateTime],
+                                   acquisitionDate: Option[DateTime],
+                                   improvementsAfterTaxStarted: Double,
+                                   claimingPRR: Boolean,
+                                   daysClaimed: Double,
+                                   daysClaimedAfter: Double): Action[AnyContent] = Action.async { implicit request =>
+
+    val totalGainModel = calculateTotalGainModel(disposalValue,
+                                                 disposalCosts,
+                                                 acquisitionValue,
+                                                 acquisitionCosts,
+                                                 improvements,
+                                                 rebasedValue,
+                                                 rebasedCosts,
+                                                 disposalDate,
+                                                 acquisitionDate,
+                                                 improvementsAfterTaxStarted)
+
+    def flatModel(): GainsAfterPRRModel = {
+      val flatPRR = acquisitionDate match {
```

Comment:
It's not necessary for working with our front end, however matching the disposal date as a Some(_) will avoid None.get's